gitlab/__init__.py

```diff
@@ -346,9 +350,17 @@ def _construct_url(self, id_, obj, parameters, action=None):
             return url
 
     def _set_auth_info(self):
-        if self.private_token and self.oauth_token:
+        if (
+            self.private_token
+            and self.oauth_token
+            or self.private_token
+            and self.job_token
+            or self.oauth_token
+            and self.job_token
+        ):
```

Comment:
This seems confusing. 

What about a true XOR?

```python
if sum(bool(arg) for arg in [self.private_token, self.oauth_token, self.job_token]) == 1
```

Reply:
Done (it's `!= 1`).

Reply:
ah, you're right!